parlai/core/logging_utils.py

```
@@ -0,0 +1,132 @@
+            self.fileHandler = logging.FileHandler(filename)
+            self.fileHandler.level = (
+                file_level
+            )  # Log to file levels: file_level and above
+            self.fileFormatter = logging.Formatter(file_format)
+            self.fileHandler.setFormatter(self.fileFormatter)
+            super().addHandler(self.fileHandler)
+
+        # Logging to stdout
+        self.streamHandler = logging.StreamHandler(sys.stdout)
+        self.streamHandler.level = (
+            console_level
+        )  # Log to stdout levels: console_level and above
+        if console_format is not None:
+            self.consoleFormatter = logging.Formatter(console_format)
+            self.streamHandler.setFormatter(self.consoleFormatter)
+        super().addHandler(self.streamHandler)
+
+        # To be used with testing_utils.capture_output()
+        self.altStream = None
+
+    def log(self, msg, level=INFO):
+        """ Default Logging function """
+        super().log(level, msg)
+
+    def add_file_handler(self, filename, level=INFO, format=None):
+        """
+        Add a file handler to the logger object
+        Use case: When logging using the logger object instead of instantiating a new ParlaiLogger
+                  this function might  be useful to add a filehandler on the go
+        Only does so if there is no file handler existing
+        """
+        if not hasattr(self, 'fileHandler'):
+            if format is None:
+                file_format = '%(asctime)s :: %(levelname)s :: %(message)s'
+            self.fileHandler = logging.FileHandler(filename)
+            self.fileHandler.level = level  # Log to file levels: level and above
+            self.fileFormatter = logging.Formatter(file_format)
+            self.fileHandler.setFormatter(self.fileFormatter)
+            super().addHandler(self.fileHandler)
+        else:
+            self.info("ParlaiLogger: A filehandler already exists")
```

do you think this should raise an exception instead?